src/app/services/edition-data.service.ts

```diff
@@ -30,11 +32,49 @@ export class EditionDataService {
 
         return editionDoc;
       }),
+      mergeMap((editionData) => this.loadXIinclude(editionData, editionUrl.substring(0, editionUrl.lastIndexOf('/') + 1))),
```

Comment:
Check edition doc type and fix `typeof source !== 'object' && typeof source === 'string'`